components/st-microelectronics/generate-ioc.stanza

```diff
@@ -0,0 +1,351 @@
+      content = append(content, to-string("Mcu.Package=%_\n" % [property(i.stm32-mfg-package)]))
+    var index:Int = 0
+    add(ip-names, RCCIP)        
+    add(ip-names, SYSIP)        
+    add(ip-names, NVICIP)        
+    ; let's list IPs now
+    for p in pins(i) do :
+      if has-property?(p.stm32-ip-name) :
+        add(ip-names, property(p.stm32-ip-name))
+    for (ip in ip-names, i in 0 to false) do :
+      ip-content = append(ip-content, to-string("Mcu.IP%_=%_\n" % [i, stm32-ip-name(ip)]))
+    ip-content = append(ip-content, to-string("Mcu.IPNb=%_\n" % [length(ip-names)]))
+    content = append(content, ip-content)
+    for p in pins(i) do :
+      if not has-property?(p.power-pin) :
+        if has-property?(p.stm32-pin-name) :
+          if prefix?(property(p.stm32-pin-name), "P") :
+            if connected?(p) :
+              content = append(content, to-string(pin-template % [index, property(p.stm32-pin-name)]))
+              index = index + 1
+    ; if not using a timer for the timebase source then the following is included
+    content = append(content, to-string("Mcu.Pin%_=VP_SYS_VS_Systick\n" % [index])) ; this is some kind of default "pin" and always defined
+    index = index + 1
+    content = append(content, to-string("Mcu.PinsNb=%_\n" % [index]))
+    ; if using a timer for the timebase then the statement should be changed to VP_SYS_VS_timX where X is the timer being utilized
+    ; don't know how to capture that yet so we default to the SysTick source
+    ; some extra pins if we are using the RTC + Calendar include the following:
+    ; Mcu.PinXX=VP_RTC_VS_RTC_Activate
+    ; Mcu.PinXX=VP_RTC_VS_RTC_Calendar
+    ; not sure how to include them either
+    for p in pins(i) do :
+      if not has-property?(p.power-pin) :
+        if has-property?(p.stm32-pin-name) :
+          if prefix?(property(p.stm32-pin-name), "P") :
+            if connected?(p) :
+              pin-content = ""
+              pin-params = ""
+              exist-params = false
+              pin-params = append(pin-params, to-string("%_.GPIOParameters=" % [property(p.stm32-pin-name)]))
+              if has-property?(p.stm32-gpio-iomode) :
+                pin-content = append(pin-content, to-string("%_.Signal=%_\n" % [property(p.stm32-pin-name), stm32-gpio-mode(property(p.stm32-gpio-iomode))]))
+              else if has-property?(p.stm32-signal-name) :
+                pin-content = append(pin-content, to-string("%_.Signal=%_\n" % [property(p.stm32-pin-name), property(p.stm32-signal-name)]))
+                ; special case of signals with names and also used as ADC
+                if has-property?(p.stm32-adc-mode) :
+                  pin-content = append(pin-content, to-string("%_.Mode=%_\n" % [property(p.stm32-pin-name), property(p.stm32-adc-mode)]))
+              if has-property?(p.stm32-gpio-speed) :
+                if exist-params == true :
+                  pin-params = append(pin-params, to-string(","))
```

Comment:
I like how you're checking `has-property?`.

Small details:
- Instead of `if boolean == true`, you can write `if boolean`
- Strings like "," and "GPIO_PuPd" without string formats (%_) don't need `to-string`. They're already Strings.

Reply:
Done.